src/utils/hooks/use-will-unmount.ts

```diff
@@ -0,0 +1,6 @@
+import { useEffect } from 'react';
+
+export function useWillUnmount(callback: EmptyFn) {
```

Comment:
So the issue I see here is that type called `EmptyFn` but the actual callback will rarely be "empty" - it will definitely do something :) Let's consider either using another type here or renaming EmptyFn to something more generic. What do you think?

Reply:
I understand the motivation for why it was used here, just commented that naming is slightly confusing in such cases. But let's leave it as is for now